client/components/PhotoCarousel.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PhotoEntry from './PhotoEntry.jsx';
+
+class PhotoCarousel extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      modalView: false,
+      thumbnails: [],
+    };
+  }
+
+  componentDidMount() {
+    fetch(`http://localhost:3333/api/photos/${this.props.id}`)
+      .then(response => response.json())
+      .then(links => {
+        return links.map(({url}) => url)
+      })
+      .then(thumbnails => this.setState({thumbnails}));
+  }
+
+  render() {
```

Comment:
This is the first time I have seen fetch method. So much easier than using an ajax request. Maybe you can walk me through how to use .then's instead of having callbacks. Great jon!

Reply:
It's a JS native method! Here's a good MDN article on using it: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch